### PR TITLE
test negative user with invalid token

### PR DESCRIPTION
**Pull request by pondrejk**
Test for BZ#2231814, I created this in coordination with dev to be used for early testing of the upcoming upstream PR with the fix. Therefore it won't pass against the current stream

### CODE REVIEW

File: tests/foreman/cli/test_user.py [hunk 1]
```diff
@@ -545,3 +545,46 @@ def test_custom_personal_access_token_role(self, target_sat):
...
+        :expectedresults: token is not created with invalid or past expire time
+
+        :CaseLevel: System
+
+        :CaseImportance: Medium
+
+        :BZ: 2231814
+        """
+        user = make_user()
+        User.add_role({'login': user['login'], 'role': 'Viewer'})
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_user.py:568]: As you already calling target_sat fixture, can you update this to use robottelo cli and cli_factory?

File: tests/foreman/cli/test_user.py [hunk 2]
```diff
@@ -545,3 +545,46 @@ def test_custom_personal_access_token_role(self, target_sat):
...
+                        'expires-at': datetime_expire,
+                    },
+                )
+        # check for past datetime
+        datetime_now = datetime.datetime.utcnow()
+        datetime_expire = datetime_now - datetime.timedelta(seconds=20)
+        datetime_expire = datetime_expire.strftime("%Y-%m-%d %H:%M:%S")
+        with pytest.raises(CLIReturnCodeError):
+            User.access_token(
+                action="create",
```
Comment (Gauravtalreja1) [inline tests/foreman/cli/test_user.py:588]:
```suggestion
        datetime_expire = datetime_expire.strftime('%Y-%m-%d %H:%M:%S')
        with pytest.raises(CLIReturnCodeError):
            User.access_token(
                action='create',
```

**Comment by ogajduse** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_user.py::TestPersonalAccessToken::test_negative_personal_access_token_invalid_date

**Comment by ogajduse** [comment]
> Do you really want to use `api` in a `cli` test module?

forgot to point to the code:
https://github.com/SatelliteQE/robottelo/pull/12300/files#diff-10a8da2d3ed98a6ca7a648dfd37ce8dc5a8aa102c3bbd0a0d181d4014f1cc362R587-R590

**Comment by pondrejk** [comment]
> Do you really want to use `api` in a `cli` test module?

[facepalm] fixed, thanks

**Review by ogajduse — APPROVED** [review]
:heavy_check_mark:

File: tests/foreman/cli/test_user.py (hunk 2 above)
Comment (ogajduse) [reply tests/foreman/cli/test_user.py]: changed

**Comment by ogajduse** [comment]
All comments are addressed, merging.

**Pull request merged**